README.md

```diff
@@ -1,77 +1,71 @@
+  </a>
+  <a href="https://github.com/lucasgpulcinelli/goICMCsim/issues?utf8=✓&q=is%3Aissue+is%3Aopen+label%3Abug">
+    <img src="https://img.shields.io/github/issues/lucasgpulcinelli/goICMCsim/bug.svg">
+  </a>
+  <a href="https://github.com/lucasgpulcinelli/goICMCsim/releases">
+    <img src="https://img.shields.io/github/v/release/lucasgpulcinelli/goICMCsim"/>
+  </a>
+  <img src="https://img.shields.io/github/license/lucasgpulcinelli/goICMCsim"/>
+</p>
+
+## Overview 📝
+This program is a simulator for the ICMC architecture (defined [here](https://github.com/simoesusp/Processador-ICMC/)). It features several upgraded functionalities compared to the C++ simulator, including:
+
+- A resizable window and fullscreen capability.
+- An instruction scroll to view all instructions and data being modified in real-time.
+- Ability to edit the stack pointer and program counter directly.
+- Enhanced error handling: the simulator will halt and indicate errors to the programmer.
+- Improved parsing of MIF files, adhering strictly to syntax definition and providing detailed error messages.
+- Capability to change character mapping MIF during runtime (without resetting).
+- Shortcuts that do not rely on keys that may not be present on laptop keyboards (e.g., insert, home, and end keys).
+- Support for Windows, macOS, and Linux.
+
+## Installation 💻
+If you prefer not to compile anything, you can download a precompiled binary for your system from the [releases page](https://github.com/lucasgpulcinelli/goICMCsim/releases).
+
+## Usage 🚀
+To get started, add a program to run and test it. You can specify MIF files in the ICMC architecture format in the command line or use the file -> open code/char MIF menu. Always specify a char MIF to ensure proper output of characters.
+
+## How to Compile from Source Code 🛠️
+1. Install a recent version of Go (at least 1.13) from [here](https://go.dev/doc/install).
+2. Install Git and a C compiler (on Windows, use MinGW).
+3. On Debian/Ubuntu-based systems, install `libgl1-mesa-dev xorg-dev`; on Fedora and Red Hat-based systems, install `libX11-devel libXcursor-devel libXrandr-devel libXinerama-devel mesa-libGL-devel libXi-devel libXxf86vm-devel`.
+4. Clone the repository and navigate to the project directory.
+5. Run `go build .` to compile and `./goICMCsim` to start an empty processor. Use `--help` to see command line options.
+6. Optionally, you can install directly into `$GOPATH/bin` with `go install github.com/lucasgpulcinelli/goICMCsim@latest`.
+
+## How to Add/Modify Instructions in the Simulator ⚙️
+To add or modify instructions:
+1. Choose an opcode for your instruction.
+2. Add it to the constants list in `processor/Instruction.go`.
+3. Add your instruction data to the `AllInstructions` list in the same file, including the opcode, mnemonic string, instruction size, and execution function.
+4. Implement the execution function. See the example in the documentation for details.
```

Comment:
Essa documentação está em algum lugar? Eu gosto da ideia de deixar bem explicadinho esses passos para criar uma nova instrução porque é algo que será necessário conforme o processador se tornar mais complexo com o passar do tempo.

Reply:
Ainda não tem a Documentação, só aquela que o Simões usava em sala. Também tem que fazer um exemplo.
Por enquanto dá pra deixar só listado os passos e que tinha listado no README atual e futuramente fazer dessa forma.

Reply:
# How to add/modify instructions in the simulator
1. Choose an opcode for your instruction.
2. Add it to the constants list at [processor/Instruction.go](processor/Instruction.go).
3. Right below, in the same file, add your instruction data to the AllInstructions list so it can be found and executed by the simulator.
4. You will need to add four pieces of information:
    - The opcode you just created.
    - An instruction to generate its mnemonic string (if your instruction receives a list of registers, just use genRegM as most instructions, or create a function yourself and add it there).
    - The instruction size in 16-bit words.
    - And a function to execute it.
5. To make the execution function, you will need a function that takes the processor context and returns an error (usually nil, to indicate everything went right).

## An instruction example
Let's create a new instruction, called `incmod`, that takes a register and adds 1 to it, but if it becomes greater than or equal to another register, it wraps around in the same way a mod would. Basically, `incmod rx, ry` is equal to `rx = (rx+1) % ry`.
This instruction is going to have opcode 0b111111, plus three bits for the first register, and three more for the second, resulting in 0b111111xxxyyydddd (where x is the first bits for the first register, y is for the second, and d is a don't care value, meaning an unused 0 or 1).

The entry in AllInstructions will be `{OpINCMOD, genRegM("incmod", 2), 1, execINCMOD},`, where OpINCMOD is defined above in the constant block as `OpINCMOD = 0b111111`. `genRegM` is used with those arguments to define the mnemonic and to say that it has two register opcodes in the usual position. 1 is to say that a single word is necessary to encode the instruction. execINCMOD is defined as follows:

```golang
func execINCMOD(pr *ICMCProcessor) error {
    // get the instruction binary data
    inst := pr.Data[pr.PC]

    // remember, those are indices from 0-7, not the actual values
    rx_index := getRegAt(inst, 7) // 7 is the first bit from right to left encoding the register index for rx
    ry_index := getRegAt(inst, 4) // same for ry

    // set the value in rx to the result of our operation
    pr.GPRRegs[rx_index] = (pr.GPRRegs[rx_index] + 1) % pr.GPRRegs[ry_index]

    // no errors happend
    return nil
}
```

Reply:
A gente cria um arquivo pra servir de documentação e coloca essa parte de adicionar uma instrução lá. E vai melhorando essa documentação com o tempo.

Reply:
O que eu vi em repositórios grandes é q eles tem um repositório só pra documentação. Acho q é pra quando alguém clonar não clonar a documentação junto.